python/cudf/cudf/core/column/column.py

```diff
@@ -685,15 +685,32 @@ def _check_scatter_key_length(
                 f"{num_keys}"
             )
 
+    def _validate_fillna_value(
+        self, fill_value: ScalarLike | ColumnLike
+    ) -> cudf.Scalar | ColumnBase:
+        """Align fill_value for .fillna based on column type."""
+        if is_scalar(fill_value):
+            return cudf.Scalar(fill_value, dtype=self.dtype)
+        return as_column(fill_value)
+
     def fillna(
         self,
-        fill_value: Any = None,
-        method: Optional[str] = None,
+        fill_value: ScalarLike | ColumnLike,
+        method: Literal["ffill", "bfill", None] = None,
     ) -> Self:
         """Fill null values with ``value``.
 
         Returns a copy with null filled.
         """
+        if not self.has_nulls(include_nan=True):
+            return self.copy()
```

Comment:
*question*: It seems like this is encoding pandas-specific behavioural matching (`nan` is `null`) in the column implementation. Do we want that?

I guess from the unchanged code below, this is not changing current behaviour.

Reply:
Yeah I think from a pure cudf API perspective this should only replace `NA` with the fill value instead of `NA` and `NaN` like pandas i.e. in the future we should probably use/check `mode.pandas_compatible` here.